api/v1alpha5/conversion.go

```diff
@@ -554,3 +566,58 @@ func Convert_v1alpha5_Bastion_To_v1alpha8_Bastion(in *Bastion, out *infrav1.Bast
 	in.Instance.FloatingIP = out.FloatingIP
 	return nil
 }
+
+func Convert_v1alpha8_SecurityGroupStatus_To_v1alpha5_SecurityGroup(in *infrav1.SecurityGroupStatus, out *SecurityGroup, s conversion.Scope) error { //nolint:revive
+	out.ID = in.ID
+	out.Name = in.Name
+	out.Rules = make([]SecurityGroupRule, len(in.Rules))
+	for i, rule := range in.Rules {
+		out.Rules[i] = SecurityGroupRule{
```

Comment:
Couldn't this be written as ?
```suggestion
		out.Rules[i] = SecurityGroupRule{
			ID:                      rule.ID,
			Direction:          rule.Direction,
			Description:      pointer.StringDeref(rule.Description, ""),
			....
```

On a related out of scope note, we should probably upgrade k8s.io/utils and start using k8s.io/utils/ptr that uses generics instead

Reply:
yeah, I started doing it and it's a big change. Here is an issue to track it: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1896